compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -159,11 +200,12 @@ private void performCodeGen(boolean shrink) {
         // collect compilation diagnostics
         List<Diagnostic> moduleDiagnostics = new ArrayList<>();
         for (ModuleContext moduleContext : pkgResolution.topologicallySortedModuleList()) {
+            Project project = moduleContext.project();
             if (moduleContext.moduleId().packageId().equals(packageContext.packageId())) {
                 if (packageCompilation.diagnosticResult().hasErrors()) {
```

Comment:
Since we are changing these parts anyway shall we merge these conditions as well
```suggestion
            if (moduleContext.moduleId().packageId().equals(packageContext.packageId()) &&
                if (packageCompilation.diagnosticResult().hasErrors()) {
```

Reply:
Fixed in c8c56fff75d0077964eb8a793986bb68f3e8a915.